### PR TITLE
Fix issues with column matching for multiple breakouts of the same column

### PR DESCRIPTION
**Pull request by ranquild**
Epic https://github.com/metabase/metabase/issues/46536

Issues I found so far:
- Incorrect parsing of legacy field refs in `findColumnIndexesFromLegacyRefs` that caused `Multiple plausible matches` error. Because field refs were parsed as `{:temporal-unit "month"}` and not `{:temporal-unit :month}`.
- Not handling for binning in `find-matching-column`.
- Issues with comparing binning values. We should only compare binning strategy-specific properties. This is the QP returning stuff like `{:strategy :bin-width :bin-width 10 :num-bins 3}` where it shouldn't.

### CODE REVIEW

File: frontend/src/metabase-lib/comparison.unit.spec.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import * as Lib from "metabase-lib";
+
+import { createQueryWithClauses } from "./test-helpers";
+
+describe("findColumnIndexesFromLegacyRefs", () => {
```
Comment (ranquild) [inline frontend/src/metabase-lib/comparison.unit.spec.ts:5]: This is to test `js.cljs` changes

File: src/metabase/lib/equality.cljc [hunk 2]
```diff
@@ -188,6 +189,19 @@
...
 
+(mu/defn- disambiguate-matches-find-match-with-same-binning :- [:maybe ::lib.schema.metadata/column]
+  "If there are multiple matching columns and `a-ref` has a binning value, check if only one column has that same
+  binning."
+  [a-ref   :- ::lib.schema.ref/ref
+    columns :- [:sequential {:min 2} ::lib.schema.metadata/column]]
+  (or (when-let [binning (lib.binning/binning a-ref)]
+        (let [matching-columns (filter (fn [col]
+                                         (lib.binning/binning= (lib.binning/binning col) binning))
+                                       columns)]
```
Comment (bshepherdson) [inline src/metabase/lib/equality.cljc:200]:
```suggestion
        (let [matching-columns (filter #(-> % lib.binning/binning (lib.binning/binning= binning))
                                       columns)]
```

File: src/metabase/lib/js.cljs [hunk 3]
```diff
@@ -1238,10 +1238,9 @@
         clj->js)))
 
 (defn- legacy-ref->pMBQL [a-legacy-ref]
-  (-> a-legacy-ref
-      (js->clj :keywordize-keys true)
-      (update 0 keyword)
-      lib.convert/->pMBQL))
+  (->> (-> a-legacy-ref (js->clj :keywordize-keys true) (update 0 keyword))
+       (mbql.normalize/normalize-fragment nil)
+       (lib.convert/->pMBQL)))
```
Comment (bshepherdson) [inline src/metabase/lib/js.cljs:1243]:
```suggestion
  (-> a-legacy-ref
      (js->clj :keywordize-keys true)
      (update 0 keyword)
      (->> (mbql.normalize/normalize-fragment nil))
      lib.convert/->pMBQL))
```

*Embed `->>` into a `->` with this one weird trick!*

It works with `as->` too if the threaded value needs to go somewhere more complicated:
```clojure
(-> ...
    (as-> $foo (some-fn (calls 1 $foo 3) 4))
    ...)
```

Also, no parens around simple function calls in `->` (like `lib.convert/->pMBQL` here).

File: test/metabase/lib/binning_test.cljc [hunk 4]
```diff
@@ -65,3 +65,24 @@
...
+      true {:strategy :num-bins :num-bins 10 :bin-width 3} {:strategy :num-bins :num-bins 10 :bin-width 4}
+      true {:strategy :num-bins :num-bins 10} {:strategy :num-bins :num-bins 10 :metadata-fn (fn [] nil)}
+      false {:strategy :num-bins :num-bins 10} {:strategy :num-bins :num-bins 20}))
+  (testing "should compare 'bin-width' binning values"
+    (are [expected x y] (= expected (lib.binning/binning= x y) (lib.binning/binning= y x))
+      true {:strategy :bin-width :bin-width 10} {:strategy :bin-width :bin-width 10}
+      true {:strategy :bin-width :bin-width 10 :num-bins 3} {:strategy :bin-width :bin-width 10}
+      true {:strategy :bin-width :bin-width 10 :num-bins 3} {:strategy :bin-width :bin-width 10 :num-bins 4}
+      true {:strategy :bin-width :bin-width 10} {:strategy :bin-width :bin-width 10 :metadata-fn (fn [] nil)}
+      false {:strategy :bin-width :bin-width 10} {:strategy :bin-width :bin-width 20})))
```
Comment (bshepherdson) [inline test/metabase/lib/binning_test.cljc:88]: <i class='graphite__hidden'>[Re: lines +71 to +89]<br/><br/></i>

Generally we follow the convention not to use commas except for separating the key-value pairs of a one-line map, see the [style guide](https://github.com/bbatsov/clojure-style-guide?tab=readme-ov-file#optional-commas-in-map-literals).<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/metabase/metabase/46979?utm_source=unchanged-line-comment">Graphite</a>.</span>

Comment (bshepherdson) [reply test/metabase/lib/binning_test.cljc:88]: Also I think this would be easier to read if the three columns were aligned.

**Review by bshepherdson — CHANGES_REQUESTED** [review]

File: test/metabase/lib/binning_test.cljc (hunk 4 above)
Comment (ranquild) [reply test/metabase/lib/binning_test.cljc]: Fixed

**Comment by ranquild** [comment]
@bshepherdson please take a look again

**Review by bshepherdson — APPROVED** [review]

**Pull request merged**